snakesolver.py

```diff
@@ -189,7 +189,7 @@ def __create_volume_flags(dimensions, index=0):
                  for i in xrange(dimensions[index])]
     
     def __repr__(self):
-        return repr(cube_flags)
+        return "\n".join([repr(x) for x in self.flags])
```

Comment:
Good catch.

Since `self.flags` may have any number of dimensions, I would prefer:

~~~python
return repr(self.flags)
~~~

(this was my first intention)